### PR TITLE
MM-23567: Remove goroutines from session_store.go

### PR DESCRIPTION
**Pull request by agnivade**
#### Summary

There isn't much benefit of running these 2 queries concurrently
versus just serially. These queries are not in the hot path anyways.
Maybe just simpler to keep it serial.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-23567

### CODE REVIEW

**Comment by agnivade** [comment]
Let me know if this makes sense. I can link it to a jira ticket and assign QA then.

**Comment by jespino** [comment]
I'm ok with that, any concern here @grundleborg or @streamer45?

**Comment by grundleborg** [comment]
@agnivade are you sure these aren't on a hot path? At a high level, the session code is about as hot as it gets, but perhaps that doesn't in practice apply to these store functions? If you're sure, then no objection to this change from me.

**Comment by streamer45** [comment]
> @agnivade are you sure these aren't on a hot path? At a high level, the session code is about as hot as it gets, but perhaps that doesn't in practice apply to these store functions? If you're sure, then no objection to this change from me.

|  Method  |  Count  |  
|---|---|
|  SessionStore.Get  |  572196 | 
|  SessionStore.Get  |  401842 | 
|  SessionStore.Get  |  144041 | 
|  SessionStore.Get  | 5966 | 
|  SessionStore.Get  | 5611 | 
|  SessionStore.Get  | 1000 |

**Comment by streamer45** [comment]
> @agnivade are you sure these aren't on a hot path? At a high level, the session code is about as hot as it gets, but perhaps that doesn't in practice apply to these store functions? If you're sure, then no objection to this change from me.

Looking at the available data it looks like those two calls are not particularly hot.

Last 30 days:

| Method | Count  |
|--------|--------|
|SessionStore.Get|575852|
|SessionStore.UpdateLastActivityAt|404173|
|SessionStore.GetSessionsWithActiveDeviceIds|145359|
|**SessionStore.GetSessions**|6016|
|SessionStore.UpdateDeviceId|5658|
|**SessionStore.Save**|1015|
|SessionStore.Remove|343|
|SessionStore.Cleanup|55|

If in doubt we could always run a load-test before merging this to gain more confidence.

**Comment by agnivade** [comment]
Yeah, to be fully confident, we can run a load test. I just sent it without even creating a JIRA ticket, so it's not urgent for me. Probably I will run a test when I have some time.

**Comment by grundleborg** [comment]
Claudio's table is good enough evidence for me!

But a load test never hurts with these things.

File: store/sqlstore/session_store.go [hunk 1]
```diff
@@ -48,29 +48,19 @@ func (me SqlSessionStore) Save(session *model.Session) (*model.Session, *model.A
...
-
 	if err := me.GetMaster().Insert(session); err != nil {
 		return nil, model.NewAppError("SqlSessionStore.Save", "store.sql_session.save.app_error", nil, "id="+session.Id+", "+err.Error(), http.StatusInternalServerError)
 	}
 
-	rtcs := <-tcs
-
-	if rtcs.Err != nil {
-		return nil, model.NewAppError("SqlSessionStore.Save", "store.sql_session.save.app_error", nil, "id="+session.Id+", "+rtcs.Err.Error(), http.StatusInternalServerError)
+	teams, err := me.Team().GetTeamsForUser(session.UserId)
```
Comment (jespino) [inline store/sqlstore/session_store.go]: this variable should be `teamMebers` to be more correct.

File: store/sqlstore/session_store.go [hunk 2]
```diff
@@ -105,25 +95,18 @@ func (me SqlSessionStore) Get(sessionIdOrToken string) (*model.Session, *model.A
...
-		close(tcs)
-	}()
 	if _, err := me.GetReplica().Select(&sessions, "SELECT * FROM Sessions WHERE UserId = :UserId ORDER BY LastActivityAt DESC", map[string]interface{}{"UserId": userId}); err != nil {
 		return nil, model.NewAppError("SqlSessionStore.GetSessions", "store.sql_session.get_sessions.app_error", nil, err.Error(), http.StatusInternalServerError)
 	}
 
-	rtcs := <-tcs
-	if rtcs.Err != nil {
-		return nil, model.NewAppError("SqlSessionStore.GetSessions", "store.sql_session.get_sessions.app_error", nil, rtcs.Err.Error(), http.StatusInternalServerError)
+	teams, err := me.Team().GetTeamsForUser(userId)
```
Comment (jespino) [inline store/sqlstore/session_store.go]: ame here, `teamMembers` is more accurate

**Comment by agnivade** [comment]
Thanks, I am going to use this as a canary testing for our new simul controller. So it will sit for a while.

**Comment by agnivade** [comment]
Ran some load tests. Did not observe any difference:

sessstore

Ignore the middle graphs. The first and the last sections are relevant.

File: store/sqlstore/session_store.go (hunk 1 above)
Comment (agnivade) [reply store/sqlstore/session_store.go]: done

File: store/sqlstore/session_store.go (hunk 2 above)
Comment (agnivade) [reply store/sqlstore/session_store.go]: done

**Comment by agnivade** [comment]
@josephbaylon - This is ready for review now. Thanks.

**Pull request merged**